src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.Analyzer.cs

```diff
@@ -0,0 +1,363 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.UsePatternMatching
+{
+    internal partial class CSharpAsAndNullCheckDiagnosticAnalyzer
+    {
+        private readonly struct Analyzer
+        {
+            private readonly SemanticModel _semanticModel;
+            private readonly ILocalSymbol _localSymbol;
+            private readonly SyntaxNode _comparison;
+            private readonly SyntaxNode _operand;
+            private readonly SyntaxNode _localStatement;
+            private readonly SyntaxNode _enclosingBlock;
+            private readonly CancellationToken _cancellationToken;
+
+            public Analyzer(
+                SemanticModel semanticModel,
+                ILocalSymbol localSymbol,
+                SyntaxNode comparison,
+                SyntaxNode operand,
+                SyntaxNode localStatement,
```

Comment:
>SyntaxNode [](start = 16, length = 10)

If this is intended to only hold a `LocalStatementSyntax`, let's narrow the type.
Same thing for `enclosingBlock` if it's indeed always a block.

Reply:
It would be a wasteful cast. I think just an assert is sufficient?